### PR TITLE
FuzzLoggerCsv

### PR DESCRIPTION
**Pull request by ottimo**
I choose this CSV content:
* timestamp
* type of log
* length of data sent or received
* hex of data sent or received
* ASCII readable data or description of log

To develop this feature I start from FuzzLoggerText

---
This change is [Reviewable](https://reviewable.io/reviews/jtpereyda/boofuzz/152)

### CODE REVIEW

File: boofuzz/fuzz_logger_csv.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    def open_test_case(self, test_case_id):
+        self._print_log_msg(["open test case", "", "", "Test case "+str(test_case_id)])
+
+    def log_fail(self, description=""):
+        self._print_log_msg(["fail", "", "", description])
+
+    def log_pass(self, description=""):
+        self._print_log_msg(["pass", "", "", description])
+
+    def _print_log_msg(self, msg, indent_level=0):
```
Comment (jtpereyda) [inline boofuzz/fuzz_logger_csv.py]: Unused param `indent_level`

File: boofuzz/fuzz_logger_csv.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    """
+    Render input_bytes as ASCII-encoded hex bytes, followed by a best effort
+    utf-8 rendering.
+
+    @param input_bytes: Arbitrary bytes.
+
+    @return: Printable string.
+    """
+    #return helpers.hex_str(input_bytes) + " " + repr(bytes(input_bytes))
+    return '0x' + ''.join("{:02x}".format(ord(b)) for b in input_bytes)
```
Comment (jtpereyda) [inline boofuzz/fuzz_logger_csv.py]: Remove commented out line. Could also do
```python
return '0x' + helpers.hex_str(input_bytes)
```

File: boofuzz/fuzz_logger_csv.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import time
+import datetime
+import csv
+
+from . import helpers
+from . import ifuzz_logger_backend
+
+# from IPython import embed
+
+# init()
```
Comment (jtpereyda) [inline boofuzz/fuzz_logger_csv.py]: Remove commented out lines

File: boofuzz/fuzz_logger_csv.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+DEFAULT_HEX_TO_STR = hex_to_hexstr
+
+
+def get_time_stamp():
+    # t = time.time()
+    # s = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime(t))
+    # s += ",%03d]" % (t * 1000 % 1000)
+    # s = t.isoFormat()
+    s = datetime.datetime.utcnow().isoformat()
```
Comment (jtpereyda) [inline boofuzz/fuzz_logger_csv.py]: Nice! An ISO format sounds better than the made up one we had...

Remove commented out timestamp method.

File: unit_tests/test_fuzz_logger_csv.py [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+        s = fuzz_logger_csv.get_time_stamp()
+
+        # Then
+        self.assertRegexpMatches(s, '\d\d\d\d-\d\d-\d\d\w\d\d:\d\d:\d\d.\d*')
+
+    def test_hex_to_hexstr(self):
+        """
+        Given: A set of bytes representing a string with several lines.
+        When: Calling hex_to_hexstr
+        Then: Hex of several-line string is output first, then repr format.
```
Comment (jtpereyda) [inline unit_tests/test_fuzz_logger_csv.py]: Update Then line to reflect new expectations, e.g.
```gherkin
        Then: Output equals hex representation of several-line string with '0x' prefix.
```

File: boofuzz/fuzz_logger_csv.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
+from __future__ import print_function
+import sys
+import time
```
Comment (jtpereyda) [inline boofuzz/fuzz_logger_csv.py]: Remove unused import `time`

**Comment by jtpereyda** [comment]
Thanks again! This is a cool feature...

**Comment by ottimo** [comment]
You are welcome :)

**Pull request merged**